source/Pacioli/Pacioli.WebApi/Controllers/UserController.cs

```diff
@@ -18,39 +20,77 @@ namespace Pacioli.WebApi.Controllers
     public class UserController : ControllerBase
     {
         private readonly UserManager<User> _userManager;
+        private readonly RoleManager<IdentityRole> _roleManager;
         private readonly IConfiguration _configuration;
+        private readonly AccessTokenGenerator _accessTokenGenerator;
 
-        public UserController(UserManager<User> userManager, IConfiguration configuration)
+        public UserController(UserManager<User> userManager, IConfiguration configuration, RoleManager<IdentityRole> roleManager)
         {
             _configuration = configuration;
+            _roleManager = roleManager;
             _userManager = userManager;
+            //TODO : Move to DI.
+            _accessTokenGenerator = new AccessTokenGenerator();
+        }
+
+        //TODO : Registration should require a special one-time permission token generated by staff.
+        [HttpPost("register")]
+        public async Task<IActionResult> RegisterAsync(RegisterModel model)
```

Comment:
I am not quite clear with how the registration workflow goes so I have a few questions:

What roles are avail?
Can anybody register and pick a role?

Reply:
I think we will have to seed roles. Will have to write a test to make sure roles cannot be generated by users.

Reply:
Who chooses which role is assumed by the new user?

Reply:
I am gonna put the registration behind the admin role, which will be seeded into the `UserIdentityDbContext` and called in `ConfigureServices()` in `Startup.cs`